test/wallet-rpc-test.js

```diff
@@ -0,0 +1,338 @@
+      const receive = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, false);
+      }
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, true);
+      }
+    });
+
+    /*
+     * assert that getaddressinfo returns the same address.
+     * the gotcha would be returning an address with the incorrect
+     * hrp which would be a problem with the network the address
+     * object uses
+     */
+    it('should return the correct address', async () => {
+      const xpub = testxpub(0, network);
+      const receive = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+      const response = await wclient.execute('getaddressinfo', [receive]);
+      assert.equal(response.address, receive);
+    });
+
+    /*
+     * select the watch only wallet that was created with the xpub
+     * at account index 0', derive the same address locally
+     * and pass to getaddressinfo, assert that ismine is true,
+     * select the standard wallet that has a different hd seed
+     * and then pass the same address to getaddressinfo and
+     * assert that ismine is false
+     */
+    it('should detect owned address', async () => {
+      const xpub = testxpub(0, network);
+      const receive = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, true);
+      }
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, false);
+      }
+    });
+
+    it('should detect a p2sh address', async () => {
+      const xpub = testxpub(0, network);
+      const pubkeys = [];
+      pubkeys.push(xpub.derive(0).derive(0).publicKey);
+      pubkeys.push(xpub.derive(0).derive(1).publicKey);
+      const script = Script.fromMultisig(2, 2, pubkeys);
+
+      const address = Address.fromScript(script);
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+      assert.equal(response.isscript, true);
+      assert.equal(response.iswitness, false);
+      assert.equal(response.witness_program, undefined);
+    });
+
+    /*
+     * assert that the correct witness program is returned for a
```

Comment:
Several comments are reiterating what can be read from the code, rather that to provide some additional information that may not be apparent.

Reply:
I think better to clarify the `it` statements and then provide any additional information within each test. This is also more consistent with the codebase, as many tests do not have such descriptions. Also most multiline comments, aside from JSDocs for functions, use the format:

```js
// Check timestamp against adj time+2hours.
// If this fails we may be able to accept
// the block later.
```

Reply:
Capitalization is also often used throughout the tests, vs. all lowercase.

Reply:
Addressed 👍